src/cargo/ops/cargo_new.rs

```diff
@@ -141,28 +297,90 @@ fn mk(config: &Config, path: &Path, name: &str,
         (Some(name), None, _, None) |
         (None, None, name, None) => name,
     };
+    
+    let mut cargotoml_path_specifier = String::new();
+    
+    let mut there_was_already_a_lib = false;
+    let mut previous_lib_relpath = String::new();
+    
+    // Calculare what [lib] and [[bin]]s do we need to append to Cargo.toml
+    
+    let explicit_better_than_implicit = opts.source_files.len() > 1;
```

Comment:
So, shall we count binaries and err out if there are more than one (like with libraries)?

That leaves the following valid configurations:

* No source files detected => create library (or binary if `--bin`);
* One library source file detected;
* One binary source file detected;
* One library and one binary source files detected;

Is it OK?